File: Algorithms/Search/Hackerland Radio Transmitters.php
<?php

/*
 * Complete the 'hackerlandRadioTransmitters' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER_ARRAY x
 *  2. INTEGER k
 */

function hackerlandRadioTransmitters($x, $k) {
    # Write your code here
    sort($x);
    $i = 0;
    $n = count($x);
    $transmitters = 0;

    while ($i < $n) {
        $transmitters++;
        $loc = $x[$i] + $k;

        while ($i < $n && $x[$i] <= $loc) {
            $i++;
        }

        $loc = $x[$i - 1] + $k;

        while ($i < $n && $x[$i] <= $loc) {
            $i++;
        }
    }
    return $transmitters;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$first_multiple_input = explode(' ', rtrim(fgets(STDIN)));

$n = intval($first_multiple_input[0]);
$k = intval($first_multiple_input[1]);

$x_temp = rtrim(fgets(STDIN));
$x = array_map('intval', preg_split('/ /', $x_temp, -1, PREG_SPLIT_NO_EMPTY));

$result = hackerlandRadioTransmitters($x, $k);

fwrite($fptr, $result . "\n");
fclose($fptr);

File: Algorithms/Search/Hackerland Radio Transmitters.py
import os

#
# Complete the 'hackerlandRadioTransmitters' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER_ARRAY x
#  2. INTEGER k
#

def hackerlandRadioTransmitters(x, k):
    # Write your code here
    x.sort()
    i = 0
    n = len(x)
    transmitters = 0

    while i < n:
        transmitters += 1
        loc = x[i] + k
        while i < n and x[i] <= loc:
            i += 1

        loc = x[i - 1] + k
        while i < n and x[i] <= loc:
            i += 1

    return transmitters


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    first_multiple_input = input().rstrip().split()

    n = int(first_multiple_input[0])
    k = int(first_multiple_input[1])
    x = list(map(int, input().rstrip().split()))

    result = hackerlandRadioTransmitters(x, k)

    fptr.write(str(result) + '\n')
    fptr.close()
